src/app/routes/clientRoutes.js

```diff
@@ -99,7 +99,13 @@ module.exports = (app, keycloak) => {
     '/:slug/explore-course', '/:slug/explore-course/*', '/:slug/signup', '/signup', '/:slug/sign-in/*',
     '/sign-in/*', '/download/*', '/accountMerge/*','/:slug/accountMerge/*', '/:slug/download/*', '/certs/*', '/:slug/certs/*', '/recover/*', '/:slug/recover/*'], redirectTologgedInPage, indexPage(false))
 
-  app.all(['*/dial/:dialCode', '/dial/:dialCode'], (req, res) => res.redirect('/get/dial/' + req.params.dialCode + '?source=scan'))
+  app.all(['*/dial/:dialCode', '/dial/:dialCode'], (req, res) => {
+    if((_.get(req,'query.channel')) && ((_.get(req,'query.channel')) === envHelper.sunbird_portal_slugForProminentFilter)){
```

Comment:
No need to check this
((_.get(req,'query.channel')) === envHelper.sunbird_portal_slugForProminentFilter)

When channel present in the URL query, then take that value & use it as slug in redirect URL.

EX:
var channelSlugVal = _.get(req,'query.channel')
if(channelSlugVal) {
  res.redirect(`/'+channelSlugVal+'/get/dial/${req.params.dialCode}?source=scan'
}